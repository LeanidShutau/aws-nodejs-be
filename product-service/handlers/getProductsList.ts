import { APIGatewayProxyHandler } from 'aws-lambda';
import 'source-map-support/register';

import { findAll } from './products-db';

const headers = {
  'Content-Type': 'application/json',
  'Access-Control-Allow-Origin': '*',
  'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
  'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PATCH, PUT',
};

export const getProductsList: APIGatewayProxyHandler = async () => {
  try {
    return {
      statusCode: 200,
      body: JSON.stringify({
        products: await findAll()
      }, null, 2),
      headers,
    };
  } catch (err) {
    return {
      statusCode: 500,
      body: JSON.stringify(err, null, 2),
      headers,
    };
  }
}
