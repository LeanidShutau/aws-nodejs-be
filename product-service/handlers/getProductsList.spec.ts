import { getProductsList } from './getProductsList';

it('renders correctly', async () => {
  const result = await getProductsList(null, null, null);
  expect(result).toMatchSnapshot();
});
