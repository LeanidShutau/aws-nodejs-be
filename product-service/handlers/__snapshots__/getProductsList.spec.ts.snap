// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "body": "{
  \\"products\\": [
    {
      \\"id\\": \\"5ecff64a-16c4-4f04-a68b-9dbd9f84dd7e\\",
      \\"title\\": \\"Air can 1\\",
      \\"price\\": 25
    },
    {
      \\"id\\": \\"e71a9424-2a97-47f5-8a16-ee965ca41ca1\\",
      \\"title\\": \\"Air can 2\\",
      \\"price\\": 40
    }
  ]
}",
  "statusCode": 200,
}
`;
